subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/JavaToolchainBuildOperationsIntegrationTest.groovy

```diff
@@ -453,8 +455,8 @@ class JavaToolchainBuildOperationsIntegrationTest extends AbstractIntegrationSpe
         eventsOnCompile = toolchainEvents(":compileKotlin")
         eventsOnTest = toolchainEvents(":test")
         then:
-        if (!isAlwaysUpToDate && Jvm.current().javaVersion.java8 && GradleContextualExecuter.configCache) {
-            // For Kotlin 1.6 the compilation is not up-to-date with configuration caching when running on Java 8
+        if (!isAlwaysUpToDate && GradleContextualExecuter.configCache) {
```

Comment:
Why is this not up-to-date for version other than 8 now?

Reply:
It seems, this was a temporary problem with the implementation. I reverted this change to keep the stricter check.

Good catch!